### ISSUE TITLE
Linuxbrew formula

### ISSUE THREAD

**Issue by harjotgill** [akiyosi/goneovim#349]
Looks like this package is being published as a cask in Homebrew. It would be really great if this package was available as a Formula within Homebrew so that it can be installed on Linux as well. For instance, `neovim-qt` is available as a Formula in Homebrew, making it really convenient to install on Linux.

**Comment by HerCerM** [comment]
Agree, I use Homebrew as my package manager for both macOS and Linux

**Comment by harjotgill** [comment]
> Why was this closed?
> This is a needy request and should be OPEN.

Happy to re-open it.

**Comment by akiyosi** [comment]
Thanks for this issue.
I am not yet familiar with homebrew, but I assume the following.

```
brew install goneovim // It treats and installs goneovim as a CUI program in Linux and MacOS.

brew install --cask goneovim // This is valid only for MacOS and installs goneovim as a GUI program.
```

Also, the following formula may be helpful:

### formula for neovim-qt
https://github.com/Homebrew/homebrew-core/pull/85526/files

### formula for rust
https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/rust.rb

**Comment by harjotgill** [comment]
> Thanks for this issue. I am not yet familiar with homebrew, but I assume the following.
> 
> ```
> brew install goneovim // It treats and installs goneovim as a CUI program in Linux and MacOS.
> 
> brew install --cask goneovim // This is valid only for MacOS and installs goneovim as a GUI program.
> ```
> 
> Also, the following formula may be helpful:
> 
> ### formula for neovim-qt
> https://github.com/Homebrew/homebrew-core/pull/85526/files
> 
> ### formula for rust
> https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/rust.rb

Your assumption is correct. 
Casks are macOS applications (mostly GUI) and they are typically launched from Launchpad, Spotlight etc. 
Formulae are normal packages that are mostly launched from Terminal. They can be CLI/TUI/GUI/Services etc. 

E,g, vimr is installed as a brew cask but in addition it provides a [script](https://github.com/qvacua/vimr/blob/develop/VimR/VimR/vimr) to help launch the application from the terminal. 

For vimr, distributing through a cask is ideal as it's closely tied to macOS. On the other hand, goneovim is cross-platform and it's more appropriate for it to use a brew formula recipe instead (or in addition to Cask).
